File: tangem-card-old/src/main/java/com/tangem/tangem_card/reader/NfcReader.java
package com.tangem.tangem_card.reader;

import java.io.IOException;

public interface NfcReader {
    byte[] getId();

    void setTimeout(int timeout)throws IOException;

    int getTimeout();

    byte[] transceive(byte[] data) throws IOException;

    void ignoreTag() throws IOException;

    void notifyReadResult(boolean success);

    void connect();

    boolean isConnected();
}

File: tangemcard-android/src/main/java/com/tangem/tangemcard/android/reader/NfcReader.kt
package com.tangem.tangemcard.android.reader

import android.nfc.tech.IsoDep

data class NfcReader(
        val nfcManager: NfcManager,
        val isoDep: IsoDep
) : com.tangem.tangemcard.reader.NfcReader {
    override fun getId(): ByteArray {
        return isoDep.tag.id
    }

    override fun setTimeout(timeout: Int) {
        isoDep.timeout = timeout
    }

    override fun getTimeout(): Int {
        return isoDep.timeout
    }

    override fun transceive(data: ByteArray?): ByteArray {
        return isoDep.transceive(data)
    }

    override fun ignoreTag() {
        nfcManager.ignoreTag(isoDep.tag)
    }

    override fun notifyReadResult(success: Boolean) {
        nfcManager.notifyReadResult(success)
    }

    override fun connect() {
        val timeout = isoDep.timeout
        isoDep.connect()
        isoDep.close()
        isoDep.connect()
        isoDep.timeout = timeout
    }

}